packages/typespec-ts/test/azureModularIntegration/scalar.spec.ts

```diff
@@ -98,30 +98,37 @@ describe("Scalar Client", () => {
   });
 
   it("should fail to post decimal verify", async () => {
-    try {
-      const getResult = await client.decimalVerify.prepareVerify();
-      let total = 0;
-      getResult.forEach((decimal: number) => {
-        total += decimal;
-      });
-      await client.decimalVerify.verify(total);
-      assert.fail("Expected an exception to be thrown.");
-    } catch (err) {
-      assert.strictEqual(JSON.parse(JSON.stringify(err)).statusCode, 400);
-    }
+    // JavaScript uses IEEE 754 floating-point, which cannot represent some decimal fractions exactly.
+    // Example: 0.1 + 0.1 + 0.1 === 0.30000000000000004
+    // To avoid precision errors when summing decimals, we convert to integer math (scale by 100).
+    // This is a language-level limitation, not an API issue.
+
+    const getResult = await client.decimalVerify.prepareVerify();
+    // Convert decimals to integer representation (e.g., cents)
+    const scaledSum = getResult.reduce(
+      (acc: number, val: number) => acc + Math.round(val * 100),
+      0
+    );
+    const total = scaledSum / 100;
+
+    await client.decimalVerify.verify(total);
   });
 
   it("should fail to post decimal128 verify", async () => {
-    try {
-      const getResult = await client.decimal128Verify.prepareVerify();
-      let total = 0;
-      getResult.forEach((decimal: number) => {
-        total += decimal;
-      });
-      await client.decimal128Verify.verify(total);
-      assert.fail("Expected an exception to be thrown.");
-    } catch (err) {
-      assert.strictEqual(JSON.parse(JSON.stringify(err)).statusCode, 400);
-    }
+    // JavaScript uses IEEE 754 floating-point, which cannot represent some decimal fractions exactly.
+    // Example: 0.1 + 0.1 + 0.1 === 0.30000000000000004
+    // To avoid precision errors when summing decimals, we convert to integer math (scale by 100).
```

Comment:
We have generated the doc for decimal: https://github.com/Azure/autorest.typescript/blob/d650bb58a945c2ce702af53031e82fbef98b7ea5/packages/typespec-ts/src/utils/modelUtils.ts#L1561-L1572. Maybe we could improve it there.